amy/workshops/views.py

```diff
@@ -1597,25 +1597,31 @@ def form_valid(self, form):
         """
 
         seat_membership = form.cleaned_data["seat_membership"]
+        seat_public = form.cleaned_data["seat_public"]
         event = form.cleaned_data["event"]
         check_ihia_old = InstructorsHostIntroductionAction.check(event)
         check_afwa_old = AskForWebsiteAction.check(event)
         check_rha_old = RecruitHelpersAction.check(event)
 
         # check associated membership remaining seats and validity
         if hasattr(self, "request") and seat_membership is not None:
+            remaining = (
+                seat_membership.public_instructor_training_seats_remaining
+                if seat_public
+                else seat_membership.inhouse_instructor_training_seats_remaining
+            )
             # check number of available seats
-            if seat_membership.seats_instructor_training_remaining == 1:
+            if remaining == 1:
                 messages.warning(
                     self.request,
-                    'Membership "{}" has 0 instructor training seats'
-                    " available.".format(str(seat_membership)),
+                    f'Membership "{seat_membership}" has 0 instructor training seats '
+                    "available.",
                 )
-            if seat_membership.seats_instructor_training_remaining < 1:
+            if remaining < 1:
```

Comment:
Hmmm how does the user get in this state?

Reply:
If user adds a learner task and marks it as part of membership Instructor Training seats, then we're using calculations to check if perhaps the limits imposed on the membership are close to exceeding.